mesonbuild/build.py

```diff
@@ -570,10 +570,11 @@ def get_default_install_dir(self) -> T.Tuple[str, str]:
     def get_custom_install_dir(self) -> T.List[T.Union[str, Literal[False]]]:
         raise NotImplementedError
 
-    def get_install_dir(self) -> T.Tuple[T.List[T.Union[str, Literal[False]]], str, Literal[False]]:
+    def get_install_dir(self) -> T.Tuple[T.List[T.Union[str, Literal[False]]], T.List[T.Optional[str]], bool]:
         # Find the installation directory.
         default_install_dir, default_install_dir_name = self.get_default_install_dir()
-        outdirs = self.get_custom_install_dir()
+        outdirs: T.List[T.Union[str, Literal[False]]] = self.get_custom_install_dir()
```

Comment:
The problem is the second assignment had a different type and mypy and pyright then treat it as a union, since lists are invariant. It wouldn’t be necessary of list was covariant